app/services/charges/charge_models/graduated_service.rb

```diff
@@ -9,18 +9,18 @@ def ranges
         charge.properties.map(&:with_indifferent_access)
       end
 
-      def compute_amount(value)
+      def compute_amount
         ranges.reduce(0) do |result_amount, range|
           flat_amount = BigDecimal(range[:flat_amount])
           per_unit_amount = BigDecimal(range[:per_unit_amount])
 
           # NOTE: Add flat amount to the total
           result_amount += flat_amount unless value.zero?
 
-          units = compute_range_units(range[:from_value], range[:to_value], value)
+          units = compute_range_units(range[:from_value], range[:to_value])
           result_amount += units * per_unit_amount
 
-          # NOTE: value is between the bounds of the current range,
+          # NOTE: aggregation_result.aggregation is between the bounds of the current range,
```

Comment:
What about adding a delegate to units in the base class?

```ruby
delegate :units, to :result
```

This way we could use units in this comments as in any other place where we relies on `value`.

Reply:
Good idea!